src/prpy/tsr/tsrlibrary.py

```diff
@@ -80,18 +81,28 @@ def __call__(self, kinbody, action_name, *args, **kw_args):
         @return list of TSRChains
         """
         kinbody_name = kw_args.get('kinbody_name', None)
-        if kinbody_name is None:
+        if kinbody_name is None and kinbody is not None:
             kinbody_name = self.get_object_type(kinbody)
             logger.debug('Inferred KinBody name "%s" for TSR.', kinbody_name)
 
+        f = None
         try:
             f = self.all_factories[self.robot_name][kinbody_name][action_name]
-        except KeyError:
-            raise KeyError('There is no TSR factory registered for action "{:s}"'
-                           ' with robot "{:s}" and object "{:s}".'.format(
-                           action_name, self.robot_name, kinbody_name))
+        except KeyError, ignored:
```

Comment:
I believe the `except Exception, name:` syntax is deprecated. Please use `except KeyError as ignored:` instead. Or just `except KeyError:` if you don't plan to use the variable at all.